ReactNativeClient/lib/services/searchengine/filterParser.ts

```diff
@@ -0,0 +1,110 @@
+
+interface Term {
+  name: string;
+  value: string;
+}
```

Comment:
Whether the Term is a negation or not should be part of the structure. The interface should be:

```
interface Term {
    negation: boolean
    name: string
    value: string
}
```

Then `folder:test` would translate to `{ name: "folder", value: "test", negation: false }` and `-folder:test` would translate `{ name: "folder", value: "test", negation: true }`.

Parsing the structure at this stage will mean less duplicated code later on.